### ISSUE TITLE
Accessing Products dictionary of StoreProductQueryResult throws InvalidCastException

### ISSUE THREAD

**Issue by trympet** [microsoft/CsWinRT#603]
Accessing `IReadOnlyDictionary` members of `StoreProductQueryResult.Products` throws `InvalidCastException`.
This includes `Keys`, `Values`, `ContainsKey`, `TryGetValue` and the indexer.

Example
```c#
StoreProductQueryResult result = await storeContext.GetUserCollectionAsync(productKinds);
IReadOnlyDictionary<string, StoreProduct> products = result.Products;
if (products.ContainsKey(addonId))
{
    Console.WriteLine("Nice");
}
```

Exception message: ```Invalid cast from 'WinRT.IInspectable' to 'ABI.System.Collections.Generic.IEnumerable`1[System.Collections.Generic.KeyValuePair`2[System.String,Windows.Services.Store.StoreProduct]]'.```

Stack trace: 
```c#
   at System.Runtime.InteropServices.DynamicInterfaceCastableHelpers.IsInterfaceImplemented(IDynamicInterfaceCastable castable, RuntimeType interfaceType, Boolean throwIfNotImplemented)
   at System.Runtime.CompilerServices.CastHelpers.ChkCast_Helper(Void* toTypeHnd, Object obj)
   at System.Runtime.CompilerServices.CastHelpers.ChkCastInterface(Void* toTypeHnd, Object obj)
   at ABI.System.Collections.Generic.IReadOnlyDictionary`2.FromAbiHelper..ctor(IMapView`2 mapView, ObjectReference`1 objRef)
   at ABI.System.Collections.Generic.IReadOnlyDictionary`2.<>c__DisplayClass11_0.<_FromMapView>b__0()
   at WinRT.IWinRTObject.<>c__DisplayClass13_0.<GetOrCreateTypeHelperData>b__0(RuntimeTypeHandle type)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at WinRT.IWinRTObject.GetOrCreateTypeHelperData(RuntimeTypeHandle type, Func`1 helperDataFactory)
   at ABI.System.Collections.Generic.IReadOnlyDictionary`2._FromMapView(IWinRTObject _this)
   at ABI.System.Collections.Generic.IReadOnlyDictionary`2.global::System.Collections.Generic.IReadOnlyDictionary<K,V>.ContainsKey(K key)
```

TFM: net5.0-windows10.0.17763.0
Runtime: 5.0.0

Seems like the cast in #536 didn't resolve all issues?

**Comment by trympet** [comment]
As a temporary workaround, using the `.ToList()` extension method seems to do the trick.

```c#
StoreProductQueryResult result = await storeContext.GetUserCollectionAsync(productKinds);
IReadOnlyDictionary<string, StoreProduct> products = result.Products;
if (products.ToList().Any(kvp => kvp.Key == addonId))
{
    Console.WriteLine("Nice");
}
```

**Comment by manodasanW** [comment]
The fix for #536 hasn't yet made it into a .NET SDK servicing build, but we plan to soon.  There is a preview version of the package which you can test out by adding the below to your application's csproj to confirm that it fixes it, but you should remove it once a .NET SDK servicing build with these fixes are available.

`
<ItemGroup>
        <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref"
                            RuntimeFrameworkVersion="10.0.17763.11-preview" />
        <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref"
                            TargetingPackVersion="10.0.17763.11-preview" />
</ItemGroup>
`

**Comment by trympet** [comment]
My bad, I was not aware. 

I can indeed confirm that it works in the preview version!
